AIMS/Controllers/AssetController.cs

```diff
@@ -0,0 +1,69 @@
+            new() { {"Asset Name", "Logitech H390 USB Headset"}, {"Type", "Headset"}, {"Tag #", "HS-2001"}, {"Status", "Assigned"} },
+            new() { {"Asset Name", "Jabra Evolve 40"}, {"Type", "Headset"}, {"Tag #", "HS-2002"}, {"Status", "Available"} },
+            new() { {"Asset Name", "Plantronics Blackwire 5220"}, {"Type", "Headset"}, {"Tag #", "HS-2003"}, {"Status", "In Repair"} },
+            new() { {"Asset Name", "Logitech Zone Wired"}, {"Type", "Headset"}, {"Tag #", "HS-2004"}, {"Status", "Surveyed"} },
+            new() { {"Asset Name", "Corsair HS60 Pro"}, {"Type", "Headset"}, {"Tag #", "HS-2005"}, {"Status", "Assigned"} },
+            new() { {"Asset Name", "Sennheiser SC 165"}, {"Type", "Headset"}, {"Tag #", "HS-2006"}, {"Status", "Available"} },
+            new() { {"Asset Name", "HyperX Cloud II"}, {"Type", "Headset"}, {"Tag #", "HS-2007"}, {"Status", "In Repair"} },
+            new() { {"Asset Name", "Sony WH-CH710N"}, {"Type", "Headset"}, {"Tag #", "HS-2008"}, {"Status", "Surveyed"} },
+            new() { {"Asset Name", "Bose 700 UC"}, {"Type", "Headset"}, {"Tag #", "HS-2009"}, {"Status", "Available"} },
+            new() { {"Asset Name", "EPOS Adapt 260"}, {"Type", "Headset"}, {"Tag #", "HS-2010"}, {"Status", "Assigned"} },
+            new() { {"Asset Name", "Microsoft Office 365"}, {"Type", "Software"}, {"Tag #", "SW-3001"}, {"Status", "Assigned"} },
+            new() { {"Asset Name", "Adobe Creative Cloud"}, {"Type", "Software"}, {"Tag #", "SW-3002"}, {"Status", "Available"} },
+            new() { {"Asset Name", "AutoCAD 2024"}, {"Type", "Software"}, {"Tag #", "SW-3003"}, {"Status", "In Repair"} },
+            new() { {"Asset Name", "Visual Studio 2022 Pro"}, {"Type", "Software"}, {"Tag #", "SW-3004"}, {"Status", "Surveyed"} },
+            new() { {"Asset Name", "Slack Enterprise"}, {"Type", "Software"}, {"Tag #", "SW-3005"}, {"Status", "Available"} },
+            new() { {"Asset Name", "Zoom Pro"}, {"Type", "Software"}, {"Tag #", "SW-3006"}, {"Status", "Assigned"} },
+            new() { {"Asset Name", "Cisco WebEx"}, {"Type", "Software"}, {"Tag #", "SW-3007"}, {"Status", "Surveyed"} },
+            new() { {"Asset Name", "Notion Team Plan"}, {"Type", "Software"}, {"Tag #", "SW-3008"}, {"Status", "Available"} },
+            new() { {"Asset Name", "IntelliJ IDEA Ultimate"}, {"Type", "Software"}, {"Tag #", "SW-3009"}, {"Status", "Assigned"} },
+            new() { {"Asset Name", "Jira Software Cloud"}, {"Type", "Software"}, {"Tag #", "SW-3010"}, {"Status", "In Repair"} },
+
+
+            
+        };
+
+        // Filter by category
+        var filteredData = tableData
+            .Where(row => row["Type"].Equals(category, StringComparison.OrdinalIgnoreCase))
+            .ToList();
+
+        this.ViewData["TableHeaders"] = tableHeaders;
+        this.ViewData["FilteredData"] = filteredData;
+
+        return View("AssetDetailsComponent");
+    }
+}
```

Comment:
Are you able to clarify the reasons for duplicating this logic accross both controllers?